### ISSUE TITLE
Keep only one JSON dependency

### ISSUE THREAD

**Issue by sbernard31** [eclipse/leshan#1045]
Currently we depend on 3 JSON libraries :confounded: :
 - **GSON** which is the historic one (it should remain only in our demos)
 - **minimal-json** which replaced it because of simple API. 
 - **jackson** which is the last choice mainly because it supports `BigInteger` which is needed to support unsigned integer from LWM2M v1.1 but also because **minimal-json** is not so much active. (e.g.https://github.com/ralfstx/minimal-json/issues/88). We have also some hint which could make us thought that **Jackson** could be faster. The main drawback of jackson the lib is fat (~1,8 Mo vs ~33 ko) but I guess we will leave with it. (except if someone propose a better solution :exploding_head:)
 
 So there is a lot of Serializer/Deserializer to rewrite with Jackson to be able to remove **GSON** and **minimal-json** dependencies

**Comment by Michal-Wadowski** [comment]
While I'm waiting for CR of #1098 I will take this task.

**Comment by Michal-Wadowski** [comment]
@sbernard31 You said that we should keep GSON only in our demos. You mean subprojects with "-demo" suffix? I found in demos dependencies to leshan-core which uses minimal-json (JsonSerDes). If we want remove minimal-json/gson from non-demo projects, we have to remove it in demos also, I think.

**Comment by sbernard31** [comment]
> While I'm waiting for CR of #1098 I will take this task.

I will look at this as soon as possible. (I hope this afternoon or tomorrow)

> @sbernard31 You said that we should keep GSON only in our demos. You mean subprojects with "-demo" suffix? I found in demos dependencies to leshan-core which uses minimal-json (JsonSerDes). If we want remove minimal-json/gson from non-demo projects, we have to remove it in demos also, I think.

I guess I was not clear enough.
The Idea is to keep only one JSON dependency in Leshan. (I mean using the same library everywhere)
Currently the idea is to keep only Jackson. (please do not hesitate to provide your opinion about this choice)

In the description of this issue, I try to explain the current state and  how we come to this "crappy" situation. (historic reasons/choices which lead to this situation) 

> GSON which is the historic one (it should remain only in our demos)

"it should remain only in our demos" I didn't mean that we must do that. I wanted to say that AFAIK GSON dependency remains only in demos. (I guess my English is not so good :sweat_smile:, I will change this)


So from my point of view the task could be : 
1. Be sure that Jackson is the right choice. (If you have no opinion I guess we can go with it unless your want to investigate this)
2. Keep only 1 dependency in leshan library (I mean the reusable part of leshan), I guess this will about removing minimal-json code.
3. keep only 1 dependency in leshan demos, this will be about removing GSON code and  probably some minimal-json code too

**Comment by Michal-Wadowski** [comment]
No, I have no strong opinion about JSON library. I think the Jackson is quite popular in java, and if it has required feature (BigInteger) we can keep it.
BTW, I will update version to at least 2.12.1, because it's needed for complete #1096 task.
